omod/src/main/java/org/openmrs/module/idgen/rest/resource/IdentifierSourceResource.java

```diff
@@ -0,0 +1,297 @@
+import org.openmrs.module.webservices.rest.web.annotation.PropertyGetter;
+import org.openmrs.module.webservices.rest.web.annotation.Resource;
+import org.openmrs.module.webservices.rest.web.representation.DefaultRepresentation;
+import org.openmrs.module.webservices.rest.web.representation.FullRepresentation;
+import org.openmrs.module.webservices.rest.web.representation.RefRepresentation;
+import org.openmrs.module.webservices.rest.web.representation.Representation;
+import org.openmrs.module.webservices.rest.web.resource.impl.DelegatingCrudResource;
+import org.openmrs.module.webservices.rest.web.resource.impl.DelegatingResourceDescription;
+import org.openmrs.module.webservices.rest.web.response.ResourceDoesNotSupportOperationException;
+import org.openmrs.module.webservices.rest.web.response.ResponseException;
+
+@Resource(name = RestConstants.VERSION_1 + IdgenRestController.IDGEN_NAMESPACE + "/identifiersource", supportedClass = IdentifierSource.class, supportedOpenmrsVersions = {
+        "1.9.*", "1.10.*", "1.11.*", "1.12.*", "2.0.*" , "2.1.*" })
+public class IdentifierSourceResource extends DelegatingCrudResource<IdentifierSource>{
+	
+    @Override
+    public DelegatingResourceDescription getRepresentationDescription(Representation rep) {
+
+        DelegatingResourceDescription description = null;
+
+        if (rep instanceof RefRepresentation) {
+            description = new DelegatingResourceDescription();
+            description.addProperty("uuid");
+            description.addProperty("display");
+            description.addSelfLink();
+            return description;
+        }
+        else if (rep instanceof DefaultRepresentation || rep instanceof FullRepresentation) {
+            description = new DelegatingResourceDescription();
+            description.addProperty("uuid");
+            description.addProperty("name");
+            description.addProperty("description");
+            description.addProperty("identifierType");
+
+            // links
+            description.addSelfLink();
+            if (rep instanceof DefaultRepresentation) {
+                description.addLink("full", ".?v=" + RestConstants.REPRESENTATION_FULL);
+            }
+        }
+        return description;
+    }
+	
+    @Override
+    public DelegatingResourceDescription getCreatableProperties() {
+        DelegatingResourceDescription description = new DelegatingResourceDescription();
+        description.addProperty("name");
+        description.addProperty("description");
+        description.addProperty("identifierType");
+        return description;
+    }
+
+    @Override
+    public DelegatingResourceDescription getUpdatableProperties() throws ResourceDoesNotSupportOperationException {
+        DelegatingResourceDescription description = new DelegatingResourceDescription();
+        description.addProperty("name");
+        description.addProperty("description");
+        description.addProperty("identifierType");
```

Comment:
Looking at this https://modules-refapp.openmrs.org/openmrs/module/idgen/editIdentifierSource.form?source=1 i think identifier type is not updatable. Not so?

Reply:
True. fixed.